### PR TITLE
feat(service-portal): Dashboard and animated icons

### PR DESCRIPTION
**Pull request by disaerna**
# Service portal - Dashboard & animated icons

## What

**Service Portal**

Add Dashboard as a landing page for Service portal.
Add animated icons for both Sidebar and Dashboard cards

_Note: after refactoring service portal to Next js -> refactor the onClick event on CategoryCard parent to use the Link prop on the card_

**Island UI**

Add Icon as an option to Category Card

**Júní**

Change redirect path to service portal from /minarsidur/postholf -> /minarsidur/

## Screenshots / Gifs

Screenshot 2022-09-01 at 15 03 09

Screenshot 2022-09-01 at 15 02 46

Videos showing animations: 

Sidebar onClick animation
https://user-images.githubusercontent.com/19873770/187948282-546ac592-3a18-45a0-95b8-827ada547471.mov

Dashboard onHover animations
https://user-images.githubusercontent.com/19873770/187948297-e9a06e92-208f-4b37-90da-688a4097395d.mov

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by albinagu — APPROVED** [review]
juni files lgtm :)

File: apps/service-portal/src/components/Sidebar/ModuleNavigation.tsx [hunk 1]
```diff
@@ -21,6 +22,7 @@ interface Props {
 
 const ModuleNavigation: FC<Props> = ({ nav, onItemClick, badge }) => {
   const [expand, setExpand] = useState(false)
+  const [hover, setHover] = useState(false)
```
Comment (thordurhhh) [inline apps/service-portal/src/components/Sidebar/ModuleNavigation.tsx:25]: Not being used?

And setTimeout on l15?

Comment (disaerna) [reply apps/service-portal/src/components/Sidebar/ModuleNavigation.tsx:25]: Ah must be some junk from testing - I'll cleanup !

File: apps/service-portal/src/screens/Dashboard/Dashboard.tsx [hunk 2]
```diff
@@ -88,14 +103,94 @@ export const Dashboard: FC<{}> = () => {
...
+  )
   useEffect(() => {
     PlausiblePageviewDetail(ServicePortalPath.MinarSidurRoot)
   }, [location])
 
+  const onHover = (id: string) => {
+    //eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const a: any = id && document.getElementById(iconIdMapper(id))
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    //const b: any = a.querySelector('svg')
```
Comment (thordurhhh) [inline apps/service-portal/src/screens/Dashboard/Dashboard.tsx:127]: remove this comment? 😃

File: apps/service-portal/src/components/Sidebar/NavItem/NavItem.tsx [hunk 3]
```diff
@@ -49,20 +53,32 @@ const NavItemContent: FC<Props> = ({
...
       alignItems="center"
       justifyContent={collapsed ? 'center' : 'spaceBetween'}
       cursor={showLock ? undefined : 'pointer'}
       position="relative"
-      onClick={showLock ? (hasArray ? undefined : onClick) : onClick}
+      onClick={() => {
+        const id = icon && iconIdMapper(icon.icon)
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const a: any = id && document.getElementById(id)
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (thordurhhh) [inline apps/service-portal/src/components/Sidebar/NavItem/NavItem.tsx:78]: Could this `: any` be a `HTMLElement` type and we could get rid of the eslint disabling?

**Review by thordurhhh — APPROVED** [review]
🔥

**Pull request merged**